packages/react-server-dom-webpack/src/ReactFlightWebpackPlugin.js

```diff
@@ -328,13 +333,39 @@ export default class ReactFlightWebpackPlugin {
   resolveAllClientFiles(
     context: string,
     contextResolver: any,
+    normalResolver: any,
     fs: any,
     contextModuleFactory: any,
     callback: (
       err: null | Error,
       result?: $ReadOnlyArray<ClientReferenceDependency>,
     ) => void,
   ) {
+    function hasUseClientDirective(source: string): boolean {
+      if (source.indexOf('use client') === -1) {
+        return false;
+      }
+      let body;
+      try {
+        body = acorn.parse(source, {
```

Comment:
Seems like this file should already be parsed by Webpack somewhere...

Reply:
It’s too early because this is pre-AST. In order to feed into webpack parsing process we’d have to add each file as a dependency — but this gets us back to where we started (we don’t want them all as dependencies cause that would everything everything in the world to the build).